mlrun/feature_store/feature_set.py

```diff
@@ -571,6 +586,10 @@ def upsert_feature(name):
             step = graph.steps[step_name]
             self._add_aggregation_to_existing(aggregation)
             step.class_args["aggregates"] = list(self._aggregations.values())
+            if emit_policy and self.spec.engine == "spark":
+                # Using simple override here - we might want to consider exploding if different emit policies
+                # were used for multiple aggregations.
```

Comment:
Yeah, we should probably add a check and raise a helpful error in such a case.

Reply:
Problem is that we don't provide a simple way for the user to fix the situation. Currently it's not easy at all - you need to remove the `aggregates` step that was previously added (behind the scenes, since the user didn't really add it, it was added through the `add_aggregates` call), and re-create the aggregations from scratch. I prefer to keep it this way for now, until we provide a simple way to do such overrides (if at all necessary).